spec/support/bulk_upload/log_to_csv.rb

```diff
@@ -8,9 +8,49 @@ def initialize(log:, line_ending: "\n", col_offset: 1, overrides: {})
     @overrides = overrides
   end
 
-  def to_csv_row
+  def row_prefix
+    [nil] * col_offset
+  end
+
+  def to_2022_csv_row
+    (row_prefix + to_2022_row).flatten.join(",") + line_ending
+  end
+
+  def to_2023_csv_row(seed: nil)
+    if seed
+      row = to_2023_row.shuffle(random: Random.new(seed))
+      (row_prefix + row).flatten.join(",") + line_ending
+    else
+      (row_prefix + to_2023_row).flatten.join(",") + line_ending
+    end
+  end
+
+  def to_2023_row
+    to_2022_row + [
+      nil, # needstype,
+      nil, # location,
```

Comment:
Are we able to set these?

Reply:
unfortunately i can't add these yet. once the UPRN work is merged, which introduces these fields to the system we can start populating them

Reply:
but that doesn't include the needs type and location? Or am I missing something?

Reply:
oh yes i see now, good point. i can add those

Reply:
just added